src/core/localize.rkt

```diff
@@ -17,24 +17,24 @@
 (define *analyze-cache* (make-hash))
 (define *analyze-context* (make-parameter #f))
 
-(define (localize-on-expression expr vars cache)
-  (hash-ref! cache expr
+(define (localize-on-expression expr vars cache precision)
+  (hash-ref! cache (cons expr precision)
```

Comment:
It'd be helpful to know why the cache needs to be precision-keyed. Is it even possible to have the same expression at two precisions? Wouldn't that require different operators and thus different precisions?